task/task.go

```diff
@@ -0,0 +1,246 @@
+// Package task provides tools to manage running a process & capturing its
+// output.
+package task
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/tcsc/levity/api"
+)
+
+// ErrInvalidState indicates that an operation was attempted on a task
+// in a state not prepared for it
+var ErrInvalidState = errors.New("task in invalid state for operation")
+
+// streamReader catches the output from one of a Cmd's output streams (i.e.
+// stdout or stderr) and writes it out to a byte buffer in a Task, under
+// a write lock.
+type streamReader struct {
+	lock *sync.RWMutex
+	dst  *bytes.Buffer
+}
+
+func (r *streamReader) Write(b []byte) (n int, err error) {
+	r.lock.Lock()
+	defer r.lock.Unlock()
+	return r.dst.Write(b)
+}
+
+// Task represents a task that has been invoked by the API server.
+type Task struct {
+	lock       sync.RWMutex
+	cmd        *exec.Cmd
+	stdout     bytes.Buffer
+	stderr     bytes.Buffer
+	statusCode api.TaskStatusCode
+	exitCode   int
+	done       chan struct{}
+}
+
+// New creates (but does not start) new task
+func New(binary string, workingDir string, env map[string]string, args ...string) *Task {
+	// Construct the underlying command which will do the heavy lifting of executing
+	// the subcommand.
+	cmd := exec.Command(binary, args...)
+	cmd.Dir = workingDir
+	cmd.Env = append(os.Environ(), formatEnvironment(env)...)
```

Comment:
I would consider exposing the server environment to the started job process a potential security issue. Curious what you and others think. cc @russjones @fspmarshall

Reply:
I agree that it presents a security risk, but given that we will happily let someone run `rm -rf /` it's probably not the biggest hole this system has. 

Code changed such that the subprocess now only gets the environment supplied in the arguments.